odbc/src/api/errors.rs

```diff
@@ -235,6 +236,9 @@ impl ODBCError {
 
 impl From<mongo_odbc_core::Error> for ODBCError {
     fn from(err: mongo_odbc_core::Error) -> Self {
+        if let Some(details) = err.details() {
+            log::error!("{details:?}");
+        }
```

Comment:
This and the following is why I went the trait route. When we convert a core error to an odbc error, it's easy to get the information we want to log this way.